### PR TITLE
got beetle: fix redis connection leaks in configuration client

### PR DESCRIPTION
**Pull request by skaes**
This patch fixes redis connection leaks due to the client exiting prematurely

### CODE REVIEW

File: go/client.go [hunk 1]
```diff
@@ -356,6 +358,14 @@ func (s *ClientState) Run() error {
 	return nil
 }
 
+func (s *ClientState) closeRedisConnections() {
+	for _, rs := range s.redisSystems {
+		if rs.currentMaster != nil {
+			rs.currentMaster.Close()
+		}
```
Comment (boosty) [inline go/client.go:365]: Nitpick: This could be a method on `RedisSystem`, so that we could just write `rs.Close()` here instead..

Comment (skaes) [reply go/client.go:365]: yes. it could. it's what I had first. but then I thought: this is an unnecessary abstraction.

Comment (skaes) [reply go/client.go:365]: there's also rs.currentMaster.IsMaster() and rs.currentMaster.Role() ...

I would need to hide all of those.

Comment (skaes) [reply go/client.go:365]: to make it consistent.

Comment (boosty) [reply go/client.go:365]: Hehe, ok. It's all very little code and just in a single file, so the value of changing it is not that big. I am fine with it as it is.

**Pull request merged**